core/modules/node/node.module

```diff
@@ -1888,6 +1886,34 @@ function node_block_view($delta = '', $settings = array()) {
         }
       }
       break;
+
+    case 'empty':
+      // Check if the promoted view exists and still has the block display.
+      $view = views_get_view('promoted');
+      if ($view && $view->access('block_1')) {
```

Comment:
For cleanliness could we rename this display to just "block" instead of "block_1"?